airgen/src/lib.rs

```diff
@@ -174,12 +174,24 @@ impl<'a, T: FieldElement> ASMPILConverter<'a, T> {
         LinkDefinitionStatement {
             source: _,
             flag,
-            params,
-            to: CallableRef { instance, callable },
+            params: lhs_params,
+            to:
+                CallableRef {
+                    instance,
+                    callable,
+                    params: rhs_params,
+                },
         }: LinkDefinitionStatement<T>,
     ) -> Link<T> {
+        // Here, we care about actual mapping of columns to target op input/output.
```

Comment:
Sorry, I still don't understand. Can you explain why we overwrite the lhs params?

Reply:
I've improved the comment here.
Essentially, what matters for the lookup is the RHS, as it may have extra (write) registers, reorderings etc.
The RHS being empty is just syntactic sugar for RHS = LHS.